### PR TITLE
Update samtools modules with config logic

### PR DESCRIPTION
**Pull request by arontommi**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `<SOFTWARE>.version.txt` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: tests/software/samtools/faidx/test.yml [hunk 1]
```diff
@@ -4,5 +4,5 @@
     - samtools
     - samtools_faidx
   files:
-    - path: ./output/samtools/test_genome.fasta.fai
+    - path: ./output/samtools/genome.fasta.fai
```
Comment (rivera10) [inline tests/software/samtools/faidx/test.yml:7]: Here should look like `output/samtools/genome.fasta.fai1`

File: tests/software/samtools/index/test.yml [hunk 2]
```diff
@@ -5,4 +5,4 @@
     - samtools_index
   files:
     - path: ./output/samtools/test_paired_end.sorted.bam.bai
```
Comment (rivera10) [inline tests/software/samtools/index/test.yml:7]: Remove extra `./` in path

File: tests/software/samtools/sort/test.yml [hunk 3]
```diff
@@ -5,4 +5,4 @@
     - samtools_sort
   files:
     - path: ./output/samtools/test.bam
```
Comment (rivera10) [inline tests/software/samtools/sort/test.yml:7]: Extra `./`

File: tests/software/samtools/view/test.yml [hunk 4]
```diff
@@ -5,4 +5,4 @@
     - samtools_view
   files:
     - path: ./output/samtools/test.bam
```
Comment (rivera10) [inline tests/software/samtools/view/test.yml:7]: Extra `./`

**Review by rivera10 — CHANGES_REQUESTED** [review]

**Review by rivera10 — APPROVED** [review]
LGTM

**Comment by drpatelh** [comment]
Thanks @arontommi !

**Pull request merged**